integrationtest/src/androidTest/java/com/swiftkey/cornedbeef/CoachMarkTestCase.java

```diff
@@ -121,7 +122,28 @@ public void testOnDismissCalled() throws InterruptedException {
         
         verify(mockListener, times(1)).onDismiss();
     }
-    
+
+    /**
+     * Verify that onTimeout is called when coach mark's timeout expired
+     * @throws InterruptedException
+     */
+    public void testOnTimeoutCalled() throws InterruptedException {
+        final long timeout = 50;
+        final OnTimeoutListener mockListener = mock(OnTimeoutListener.class);
+        mCoachMark = new TestCoachMark.TestCoachMarkBuilder(
+                mActivity, mAnchor, "spam spam spam")
+                .setOnTimeoutListener(mockListener)
+                .setTimeout(timeout)
+                .build();
+
+        showCoachMark(getInstrumentation(), mCoachMark);
+
+        // Wait until timeout
+        Thread.sleep(timeout);
+
+        verify(mockListener).onTimeout();
+    }
+
     /**
```

Comment:
Adding a test with all the listeners could be useful too, so we make sure they are all still triggered.